src/tools/rbd/action/ImageMeta.cc

```diff
@@ -98,8 +98,12 @@ static int do_metadata_remove(librbd::Image& image, const char *key)
 {
   int r = image.metadata_remove(key);
   if (r < 0) {
-    std::cerr << "failed to remove metadata " << key << " of image : "
-              << cpp_strerror(r) << std::endl;
+    if( r== -ENOENT)
+        std::cerr << "rbd: no existing metadata key " << key << " of image : "
+                  << cpp_strerror(r) << std::endl;
+    else
+        std::cerr << "failed to remove metadata " << key << " of image : "
+                  << cpp_strerror(r) << std::endl;
   }
```

Comment:
> Nit: move this outside the nested if so that it doesn't print two error messages

I think @joscollin means that? without nested:
`if (r < 0 && r != -ENOENT) { `
`} else if (r == -ENOENT) { }`